src/outputs/ascent.cpp

```diff
@@ -0,0 +1,199 @@
+
+    // create the coordinate set
+    mesh["coordsets/coords/type"] = "uniform";
+
+    mesh["coordsets/coords/dims/i"] = nx + 1;
+    mesh["coordsets/coords/dims/j"] = ny + 1;
+    if (nz > 1) {
+      mesh["coordsets/coords/dims/k"] = nz + 1;
+    }
+
+    // add origin and spacing to the coordset (optional)
+    mesh["coordsets/coords/origin/x"] = corner[0];
+    mesh["coordsets/coords/origin/y"] = corner[1];
+    if (nz > 1) {
+      mesh["coordsets/coords/origin/z"] = corner[2];
+    }
+
+    mesh["coordsets/coords/spacing/dx"] = dx1;
+    mesh["coordsets/coords/spacing/dy"] = dx2;
+    if (nz > 1) {
+      mesh["coordsets/coords/spacing/dz"] = dx3;
+    }
+
+    // add the topology
+    mesh["topologies/topo/type"] = "uniform";
+    mesh["topologies/topo/coordset"] = "coords";
+
+    // indicate ghost zones with ascent_ghosts set to 1
+    Node &n_field = mesh["fields/ascent_ghosts"];
+    n_field["association"] = "element";
+    n_field["topology"] = "topo";
+    n_field["values"].set(conduit::DataType::int32(ncells));
+    conduit::int32_array vals_array = n_field["values"].value();
+
+    int idx = 0;
+    for (int k = kb.s; k <= kb.e; k++) {
+      for (int j = jb.s; j <= jb.e; j++) {
+        for (int i = ib.s; i <= ib.e; i++) {
+          if ((i < ib_int.s) || (ib_int.e < i) || (j < jb_int.s) || (jb_int.e < j) ||
+              ((nz > 1) && ((k < kb_int.s) || (kb_int.e < k)))) {
+            vals_array[idx] = 1;
+          }
+          idx++;
+        }
+      }
+    }
+
+    // create a field for each component of each variable pack
+    auto &mbd = pmb->meshblock_data.Get();
+
+    for (auto &vars : mbd->GetCellVariableVector()) {
+      const std::string packname = vars->label();
+      auto const &labels = vars->metadata().getComponentLabels();
+      auto const &data = vars->data;
+
+      for (int icomp = 0; icomp < labels.size(); ++icomp) {
+        const std::string varname = packname + ":" + labels.at(icomp);
+        mesh["fields/" + varname + "/association"] = "element";
+        mesh["fields/" + varname + "/topology"] = "topo";
+        mesh["fields/" + varname + "/values"].set_external(&data(icomp, 0, 0, 0), ncells);
+      }
+    }
+  }
+
+  // make sure we conform:
+  Node verify_info;
+  if (!conduit::blueprint::mesh::verify(root, verify_info)) {
+    if (parthenon::Globals::my_rank == 0) {
+      std::cout << "blueprint::mesh::verify failed!" << std::endl;
+    }
+    verify_info.print();
+  }
+  a.publish(root);
+
+  // setup actions
+  Node actions;
+  Node &add_act = actions.append();
+  add_act["action"] = "add_scenes";
+
+  // declare a scene (s1) with one plot (p1)
+  Node &scenes = add_act["scenes"];
+  scenes["s1/plots/p1/type"] = "pseudocolor";
+  scenes["s1/plots/p1/field"] = "advected:Advected_0_0";
+
+  // Set the output file name (ascent will add ".png")
+  scenes["s1/image_prefix"] = "ascent_render";
+
+  // execute the actions
+  a.execute(actions);
+
+  // close ascent
+  a.close();
```

Comment:
If we initializae at simulation start, this would also have to be called only at the end of a simulation.